### PR TITLE
feat: add `deno_web` APIs like `setTimeout`

### PR DESCRIPTION
**Pull request by bajtos**
Add JS APIs from `deno_web` crate.

Most notably:

- Events and `AbortController`
- `TextEncoder` and `TextDecoder` (will be needed for WASM)
- `File`, `FileReader` and `Blob`
- `MessageChannel`, `structuredClone` and friends
- `atob` and `atoa` (will be needed for WASM)
- `setTimeout`, `setInterval`, and friends
- `Performance` APIs
- Streams (`ReadableStream`, `WritableStream`, etc.)
- `URL` and friends

See #7

### CODE REVIEW

**Comment by bajtos** [comment]
@juliangruber This pull request is adding many new public APIs, significantly increasing the amount of stuff we will need to support in the future.

Most of these APIs are lightweight (e.g. AbortController, events, Performance, URL), so I don't anticipate any need to restrict their usage when we start limiting resource consumption.

Many other APIs will be needed in the future:
- `TextEncoder`, `TextDecoder` and `btoa` (or `atob`?) are required by wasm-bindgen
- Web Streams are needed for Fetch API. We will need some concept of streams anyways, I think it's best to build on a standard, even if it's not the greatest API.

Could you please review the list of the new APIs and let me know if there is anything you would prefer to remove from the global scope (make it private)?

Do you have any opinions on how to test this? I don't want to replicate the Deno test suite or Web API conformance tests. At the same time, I think it would be nice to have some sort of smoke tests to verify that the APIs are correctly exposed. But that's quite a lot of work, considering how many new APIs are added in this PR.

File: docs/building-modules.md [hunk 1]
```diff
@@ -38,12 +38,140 @@ Hello universe!
...
+- [TextDecoder](https://developer.mozilla.org/en-US/docs/Web/API/TextDecoder)
+- [TextEncoder](https://developer.mozilla.org/en-US/docs/Web/API/TextEncoder)
+- [TextDecoderStream](https://developer.mozilla.org/en-US/docs/Web/API/TextDecoderStream)
+- [TextEncoderStream](https://developer.mozilla.org/en-US/docs/Web/API/TextEncoderStream)
+
+#### File API
+
+- [Blob](https://developer.mozilla.org/en-US/docs/Web/API/blob)
+- [File](https://developer.mozilla.org/en-US/docs/Web/API/File)
+- [FileReader](https://developer.mozilla.org/en-US/docs/Web/API/FileReader)
```
Comment (bajtos) [inline docs/building-modules.md:74]: I have mixed feelings about `Blob` and `File` related APIs. 

There is no access to the file system yet, so these APIs are only building blocks that cannot be meaningfully used (yet).

I also think the filesystem abstraction is not the right solution to allow modules to persist data. Something closer to a key/value store may be a much better fit. There are also LocalStorage and IndexedDB Web APIs (see [Deno's WebStorage API](https://deno.land/manual@v1.30.3/runtime/web_storage_api) and [MDN IndexedAPI](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API)) we may want to provide.

I am proposing to remove these three APIs for now. We can easily add them later, if there is a need.

WDYT?

Comment (bajtos) [reply docs/building-modules.md]: I removed these APIs in 8579744

**Review by juliangruber — APPROVED** [review]

**Comment by bajtos** [comment]
Cool.

> What about just checking whether the global is defined? And then progress with regression tests where necessary. We could also add some more tests to one specific module, which "proves" that in general integrating modules from Deno works.

I will land this PR as-is and open another PR to add some smoke tests. I'll see if I can find a reasonably easy way for testo testn masse.

**Pull request merged**